<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainFrame</class>
 <widget class="QWidget" name="MainFrame">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>598</width>
    <height>356</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>FFTExplorer</string>
  </property>
  <layout class="QGridLayout">
   <item row="1" column="0">
    <layout class="QHBoxLayout">
     <item>
      <widget class="QPushButton" name="StartButton">
       <property name="toolTip">
        <string>Begin FFT conversion process</string>
       </property>
       <property name="text">
        <string>&amp;Start</string>
       </property>
       <property name="shortcut">
        <string>Alt+S</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="StopButton">
       <property name="toolTip">
        <string>Stop conversion process, freeze the display</string>
       </property>
       <property name="text">
        <string>S&amp;top</string>
       </property>
       <property name="shortcut">
        <string>Alt+T</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="spacer1">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>60</width>
         <height>31</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="dummyButton">
       <property name="text">
        <string>pushButton5</string>
       </property>
       <property name="shortcut">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="QuitButton">
       <property name="toolTip">
        <string>Exit FFTExplorer</string>
       </property>
       <property name="text">
        <string>&amp;Quit</string>
       </property>
       <property name="shortcut">
        <string>Alt+Q</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="0" column="0">
    <widget class="QTabWidget" name="TabPane">
     <property name="toolTip">
      <string/>
     </property>
     <property name="tabShape">
      <enum>QTabWidget::Rounded</enum>
     </property>
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="timePage">
      <attribute name="title">
       <string>Time</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="frequencyPage">
      <attribute name="title">
       <string>Frequency</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="bothPage">
      <attribute name="title">
       <string>Both</string>
      </attribute>
     </widget>
     <widget class="QWidget" name="configurationPage">
      <attribute name="title">
       <string>Configuration</string>
      </attribute>
      <layout class="QGridLayout">
       <item row="0" column="0">
        <widget class="Q3GroupBox" name="groupBox17" native="true">
         <property name="title" stdset="0">
          <string>Configuration</string>
         </property>
         <layout class="QGridLayout">
          <property name="margin">
           <number>4</number>
          </property>
          <property name="spacing">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="Q3GroupBox" name="groupBox2" native="true">
            <property name="title" stdset="0">
             <string>General</string>
            </property>
            <layout class="QGridLayout">
             <property name="margin">
              <number>4</number>
             </property>
             <property name="spacing">
              <number>4</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="textLabel9">
               <property name="text">
                <string>Sample Rate</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="sampleRateText">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>0</height>
                </size>
               </property>
               <property name="toolTip">
                <string>Set the sample rate in samples per second</string>
               </property>
              </widget>
             </item>
             <item row="0" column="3">
              <widget class="QLineEdit" name="arraySizeEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Set the FFT conversion array size</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2" rowspan="2">
              <widget class="QLabel" name="textLabel10">
               <property name="text">
                <string>FFT Array Size</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="3" rowspan="2">
              <widget class="QPushButton" name="applyButton">
               <property name="toolTip">
                <string>If this button is enabled, click it to accept your changes</string>
               </property>
               <property name="text">
                <string>Apply</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="textLabel1_3">
               <property name="text">
                <string>Moving Average Period</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="movingAverageEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Set the window size for moving averages</string>
               </property>
              </widget>
             </item>
             <item row="2" column="2">
              <widget class="QCheckBox" name="gridCheckBox">
               <property name="toolTip">
                <string>Include calibration grids in displays</string>
               </property>
               <property name="text">
                <string>Grids in charts</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item row="1" column="0">
           <layout class="QHBoxLayout">
            <item>
             <widget class="Q3ButtonGroup" name="sourceButtonGroup" native="true">
              <property name="title" stdset="0">
               <string>Source</string>
              </property>
              <property name="exclusive" stdset="0">
               <bool>true</bool>
              </property>
              <property name="radioButtonExclusive" stdset="0">
               <bool>true</bool>
              </property>
              <layout class="QHBoxLayout">
               <property name="spacing">
                <number>0</number>
               </property>
               <property name="margin">
                <number>4</number>
               </property>
               <item>
                <widget class="QRadioButton" name="internalRadioButton">
                 <property name="toolTip">
                  <string>Select the internal data source</string>
                 </property>
                 <property name="text">
                  <string>Internal</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="externalRadioButton">
                 <property name="toolTip">
                  <string>Select an external data source</string>
                 </property>
                 <property name="text">
                  <string>External</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="Q3GroupBox" name="externalSourceBox" native="true">
              <property name="title" stdset="0">
               <string>External Source</string>
              </property>
              <layout class="QHBoxLayout">
               <property name="spacing">
                <number>4</number>
               </property>
               <property name="margin">
                <number>4</number>
               </property>
               <item>
                <widget class="QLabel" name="textLabel11">
                 <property name="text">
                  <string>Path</string>
                 </property>
                 <property name="wordWrap">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="dataSourceText">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="toolTip">
                  <string>Specify a path to an external data source</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="textLabel1">
                 <property name="text">
                  <string>Format</string>
                 </property>
                 <property name="wordWrap">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="dataFormatComboBox">
                 <property name="toolTip">
                  <string>Select the format for external data</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
           </layout>
          </item>
          <item row="2" column="0">
           <widget class="Q3GroupBox" name="internalSourceBox" native="true">
            <property name="title" stdset="0">
             <string>Internal Source</string>
            </property>
            <layout class="QGridLayout">
             <property name="margin">
              <number>4</number>
             </property>
             <property name="spacing">
              <number>4</number>
             </property>
             <item row="1" column="0">
              <widget class="QLabel" name="textLabel3">
               <property name="text">
                <string>Modulation level %</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QLabel" name="textLabel1_2">
               <property name="text">
                <string>Carrier Freq. Hz</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="textLabel4">
               <property name="text">
                <string>Noise Level %</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="carrierFrequencyEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Enter the desired internal signal frequency</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="modLevelEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Enter the desired carrier modulation level</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="noiseLevelEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Enter a noise level, which is added to the internal signal</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QLabel" name="textLabel2">
               <property name="text">
                <string>Modulation Freq. Hz</string>
               </property>
               <property name="wordWrap">
                <bool>false</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="2" rowspan="2" colspan="2">
              <widget class="Q3ButtonGroup" name="modTypeButtonGroup" native="true">
               <property name="margin" stdset="0">
                <number>0</number>
               </property>
               <property name="title" stdset="0">
                <string>Modulation Type</string>
               </property>
               <property name="exclusive" stdset="0">
                <bool>true</bool>
               </property>
               <widget class="QRadioButton" name="amButton">
                <property name="geometry">
                 <rect>
                  <x>11</x>
                  <y>22</y>
                  <width>50</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="toolTip">
                 <string>Choose amplitude modulation</string>
                </property>
                <property name="text">
                 <string>AM</string>
                </property>
               </widget>
               <widget class="QRadioButton" name="fmButton">
                <property name="geometry">
                 <rect>
                  <x>70</x>
                  <y>20</y>
                  <width>60</width>
                  <height>21</height>
                 </rect>
                </property>
                <property name="toolTip">
                 <string>Choose frequency modulation</string>
                </property>
                <property name="text">
                 <string>FM</string>
                </property>
               </widget>
              </widget>
             </item>
             <item row="0" column="3">
              <widget class="QLineEdit" name="modulationFrequencyEdit">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="toolTip">
                <string>Enter a frequency by which the carrier signal is modulated</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="helpPage">
      <attribute name="title">
       <string>Help</string>
      </attribute>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <customwidgets>
  <customwidget>
   <class>Q3GroupBox</class>
   <extends>QWidget</extends>
   <header>q3groupbox.h</header>
  </customwidget>
  <customwidget>
   <class>Q3ButtonGroup</class>
   <extends>QWidget</extends>
   <header>q3buttongroup.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>StartButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>StartDisplay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>StopButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>StopDisplay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>QuitButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gridCheckBox</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dataFormatComboBox</sender>
   <signal>activated(int)</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sampleRateText</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>arraySizeEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dataSourceText</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>applyButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>internalRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>externalRadioButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>carrierFrequencyEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modLevelEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>noiseLevelEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modulationFrequencyEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>amButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fmButton</sender>
   <signal>clicked()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>carrierFrequencyEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modLevelEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>noiseLevelEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modulationFrequencyEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sampleRateText</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>arraySizeEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dataSourceText</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>movingAverageEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainFrame</receiver>
   <slot>enableApply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>movingAverageEdit</sender>
   <signal>returnPressed()</signal>
   <receiver>MainFrame</receiver>
   <slot>applyPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>20</x>
     <y>20</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
